pkg/curatedpackages/packageclient.go

```diff
@@ -46,3 +59,74 @@ func convertBundleVersionToPackageVersion(bundleVersions []packagesv1.SourceVers
 	}
 	return versions
 }
+
+func (pc *PackageClient) GeneratePackages() ([]packagesv1.Package, error) {
+	packageMap := pc.getPackageNameToPackage()
+	var packages []packagesv1.Package
+	for _, p := range pc.packages {
+		bundlePackage := packageMap[strings.ToLower(p)]
+		if bundlePackage.Name == "" {
+			fmt.Println(fmt.Errorf("unknown package %s", p).Error())
+			continue
+		}
+		name := CustomName + strings.ToLower(bundlePackage.Name)
+		packages = append(packages, convertBundlePackageToPackage(bundlePackage, name, pc.bundle.APIVersion))
+	}
+	return packages, nil
+}
+
+func (pc *PackageClient) WritePackagesToFile(packages []packagesv1.Package, d string) error {
+	directory := filepath.Join(d, packageLocation)
+	if !validations.FileExists(directory) {
+		if err := os.Mkdir(directory, dirPermission); err != nil {
+			return fmt.Errorf("unable to create directory %s", directory)
+		}
+	}
+
+	for _, pack := range packages {
+		displayPackage := NewDisplayPackage(pack)
+		content, err := yaml.Marshal(displayPackage)
+		if err != nil {
+			fmt.Println(fmt.Errorf("unable to parse package %s %v", pack.Name, err).Error())
+			continue
+		}
+		err = writeToFile(directory, pack.Name, content)
```

Comment:
Since we're writing YAML files, maybe it makes more sense to just write a single YAML file, separated using the YAML document syntax? That way we don't need a directory, don't have to necessarily create directories, and can even output to stdout (in a nice UNIX-friendly way).

Reply:
@TerryHowe and I discussed this and we both agreed that the single YAML file is not an ideal solution here. The reasoning was that, with all the generated packages, the single file will get huge very quick and will be hard to read and update.

Reply:
I don't see how one 500 line file is harder to deal with vs twenty 25 line files, but 🤷🏻